### ISSUE TITLE
[ Calculs ] Nombre complexe (simplification des résultats complémentaires)

### ISSUE THREAD

**Issue by fmOOmf** [UpsilonNumworks/Upsilon#167]
Bonjour à toute l'équipe

_(En préambule : le comportement décrit ici est similaire pour le firmware Epsilon E17)_
Dans l'application "Calculs", lorsque le résultat est un nombre complexe il est possible de consulter des résultats complémentaires en cliquant sur "...".
On a alors accès à un graphique où sont représentés de manière didactique le nombre complexe, le module, l'argument. Suivent les caractéristiques principales (module, argument parties réelle et imaginaire). 

Pour certains nombres complexes, l'argument et le module sont proposés sous des formes très compliquées (cf. copies d'écran).
Exemple : 
Dans l'application "Calculs", taper (1+i)^0.8
Résultats complémentaires attendus : 
Module = 2^0.4 = root(4,5), et argument = pi/5

Résultats affichés : 
Module : abs(√(2)×root(4,5)×√(-√(5)+5)×𝐢+root(4,5)×√(5)+root(4,5))/4
Argument : arg(√(2)×root(4,5)×√(-√(5)+5)×𝐢+root(4,5)×√(5)+root(4,5)/4)
Les formats proposés pour les parties réelles et imaginaires sont également concernés

**Evolution** proposée : 
Simplifier lorsque c'est possible les formats des formes exactes des résultats complémentaires avant affichage, ou préférer les formes approchées (~) si aucune simplification n'est trouvée.

Copies d'écran : 
Le nombre complexe de l'exemple : 
![image](https://user-images.githubusercontent.com/98671961/155027478-3d0ae688-b630-4e83-bd8f-480133720742.png)

Les résultats complémentaires proposés : 
![image](https://user-images.githubusercontent.com/98671961/155029431-7a1ad6a6-d214-4705-b8ff-be6139093937.png)

**Comment by Lauryy06** [comment]
Je pense que les résultats exacts sont importants, mais effectivement un travail peut-être fait, notamment au niveau de la simplification des racines imbriquées.

**Comment by Overengined** [comment]
> Je pense que les résultats exacts sont importants, mais effectivement un travail peut-être fait, notamment au niveau de la simplification des racines imbriquées.

I agree, it's important to preserve the precision of the results. Nevertheless, we should think of a clear and quick way to indicate that a result cannot be simplified, in order to avoid confusion. Maybe adding a small gray dot  on the upper left corner of the concerned content box ?

**Comment by fmOOmf** [comment]
Hello.

Yes you are right.
I proposed to use (~) when the result is not provided in an exact form, and (=) otherwise.

The Upsilon team will choose the solution they find most convenient.

**Comment by fmOOmf** [comment]
Je ne sais pas si cela peut aider mais sur  l'exemple (1+i)^0,8 : 
Upsilon sait trouver les valeurs exactes des modules et argument de 1+i  : **sqrt(2)** et **pi/4**.

Donc il n'est pas loin  des valeurs exactes à trouver pour (1+i)^n  il "suffirait" de faire **module^n** et **argument*n**.
Sur l'exemple :  
- module devient  sqrt(2)^0,8, ce qui donne bien l'attendu : 2^0.4 = root(4,5)
- argument devient (pi/4)*0,8, ce qui donne bien l'attendu : pi/5

Quelquefois (et c'est le cas ici) c'est la forme module et argument qui est la plus adaptée, et qui permet d'arriver au résultat sous sa forme la plus simple.
